### PR TITLE
chore(weave): Improve Prompt Notebook Flows

### PR DESCRIPTION
**Pull request by tssweeney**
Note: Uptook some changes from Stacey's https://github.com/wandb/weave/pull/639

Checklist:

* Homepage Link:
  * `/examples/prompts/llm_monitoring/openai_client_quickstart.ipynb` 
* LLM Monitoring Template Instructions:
  * `/examples/prompts/llm_monitoring/openai_client_quickstart.ipynb` (same as above)
  * `/examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb`
  * Note: moved the `dev` folder to `experimental` since the synthetic data code cannot be executed in a colab since it relies on flat files not shipped with the weave package.
* Trace Debug Template Instructions:
  * `/examples/prompts/trace_debugging/trace_quickstart_langchain.ipynb`
  * `/examples/prompts/trace_debugging/trace_quickstart_decorator.ipynb`
  * `examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb`

### CODE REVIEW

File: examples/experimental/prompts_dev/synthetic_openai_data.ipynb [hunk 1]
```diff
@@ -37,6 +45,18 @@
...
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "61b9c12a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "WB_ENTITY = \"\" # replace with your W&B username or team name\n",
```
Comment (staceysv) [inline examples/experimental/prompts_dev/synthetic_openai_data.ipynb:55]: minor/nit: I set this to WB_ENTITY = so the cell errors, WB_ENITY = "" will let the cell execute & have weird behavior

File: examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb [hunk 2]
```diff
@@ -12,18 +43,32 @@
...
    "id": "c3e49bf5-b40a-4039-85e7-4b4795c0665e",
    "metadata": {},
    "outputs": [],
    "source": [
-    "WB_ENTITY = \"timssweeney\"\n",
-    "WB_PROJECT = \"trace_quickstart\"\n",
-    "WB_STREAM = \"log_stream\"\n",
-    "\n",
-    "st = StreamTable(f\"{WB_ENTITY}/{WB_PROJECT}/{WB_STREAM}\")"
+    "WB_ENTITY = \"\" # replace with your W&B username or team name\n",
```
Comment (staceysv) [inline examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb:61]: minor/nit: looks like i missed this, set entity to null here as well?

**Comment by staceysv** [comment]
really appreciate the cosmetic stream renaming—i will make sure to rename the projects to be more descriptive too, once we fix the "first project != "weave"" bug.

File: examples/experimental/prompts_dev/synthetic_openai_data.ipynb (hunk 1 above)
Comment (tssweeney) [reply examples/experimental/prompts_dev/synthetic_openai_data.ipynb]: done

File: examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb (hunk 2 above)
Comment (tssweeney) [reply examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb]: done

File: examples/experimental/prompts_dev/synthetic_openai_data.ipynb
Comment (staceysv) [inline examples/experimental/prompts_dev/synthetic_openai_data.ipynb:1]: great, i think there are other issues here we can deal with later (like the relative file path loading for shakespeare.text)—seems fine in dev

File: examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb
Comment (staceysv) [inline examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb:1]: not sure if this errors is expected here:
- second "Use CURL" cell doesn't work for me, with entity/project/stream name correctly added: I get the following
{
    "error": {
        "message": "Only POST requests are accepted.",
        "type": "invalid_request_error",
        "param": null,
        "code": "method_not_supported"
    }
}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   183  100   183    0     0    631      0 --:--:-- --:--:-- --:--:--   633
curl: (3) URL using bad/illegal format or missing URL
bash: line 5: -H: command not found
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
[<ipython-input-10-f3a03a8c75c5>](https://localhost:8080/#) in <cell line: 1>()
----> 1 get_ipython().run_cell_magic('bash', '', 'curl "https://api.wandb.ai/proxy/openai/v1/chat/completions" \\\n-H "Authorization: Bearer $WANDB_API_KEY:$OPENAI_API_KEY" \\\n-H "Content-Type: application/json" \\\n-H "X-Wandb-Entity: stacey" \\ \n-H "X-Wandb-Project: weave" \\\n-H "X-Wandb-Stream: openai_stream" \\\n-H "X-Wandb-Client-Id: Shakespearean-Gradients-001" \\\n-H "X-Wandb-Attribute-Persona: Shakespeare" \\\n-d \'{\n     "model": "gpt-3.5-turbo",\n     "messages": [{"role": "user", "content": "Tell me a joke about loss functions!"}]\n   }\'\n')

4 frames
<decorator-gen-103> in shebang(self, line, cell)

[/usr/local/lib/python3.10/dist-packages/IPython/core/magics/script.py](https://localhost:8080/#) in shebang(self, line, cell)
    243             sys.stderr.flush()
    244         if args.raise_error and p.returncode!=0:
--> 245             raise CalledProcessError(p.returncode, cell, output=out, stderr=err)
    246 
    247     def _run_script(self, p, cell, to_close):

CalledProcessError: Command 'b'curl "https://api.wandb.ai/proxy/openai/v1/chat/completions" \\\n-H "Authorization: Bearer $WANDB_API_KEY:$OPENAI_API_KEY" \\\n-H "Content-Type: application/json" \\\n-H "X-Wandb-Entity: stacey" \\ \n-H "X-Wandb-Project: weave" \\\n-H "X-Wandb-Stream: openai_stream" \\\n-H "X-Wandb-Client-Id: Shakespearean-Gradients-001" \\\n-H "X-Wandb-Attribute-Persona: Shakespeare" \\\n-d \'{\n     "model": "gpt-3.5-turbo",\n     "messages": [{"role": "user", "content": "Tell me a joke about loss functions!"}]\n   }\'\n'' returned non-zero exit status 127.

File: examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb [hunk 3]
```diff
@@ -206,6 +272,19 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "!pip install -qqq weave tiktoken"
```
Comment (staceysv) [inline examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb:285]: minor/confusing: I get a message about errors in the installation/needing to restart the runtime, but the rest of the colab runs fine:
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pandas-gbq 0.17.9 requires pyarrow<10.0dev,>=3.0.0, but you have pyarrow 13.0.0 which is incompatible.
tensorflow 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.8.0 which is incompatible.
Successfully installed GitPython-3.1.37 aiofiles-23.2.1 aioprocessing-2.0.1 analytics-python-1.2.9 backoff-2.2.1 black-23.9.1 docker-pycreds-0.4.0 flask-cors-4.0.0 gitdb-4.0.10 gql-3.4.1 graphql-core-3.2.3 ipynbname-2023.2.0.0 janus-1.0.0 jedi-0.19.0 mypy-extensions-1.0.0 objgraph-3.6.0 pathspec-0.11.2 pathtools-0.1.2 pyarrow-13.0.0 pynndescent-0.5.10 python-json-logger-2.0.7 requests-toolbelt-0.10.1 sentry-sdk-1.28.1 setproctitle-1.3.2 smmap-5.0.1 tiktoken-0.5.1 typeguard-4.1.5 typing-extensions-4.8.0 umap-learn-0.5.4 urllib3-1.26.16 wandb-0.15.11 weave-0.28.0
WARNING: The following packages were previously imported in this runtime:
  [urllib3]
You must restart the runtime in order to use newly installed versions.

File: examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb [hunk 4]
```diff
@@ -225,18 +304,6 @@
     "r = openaimon.ChatCompletion.create(model=OPENAI_MODEL, messages=[{\"role\": \"user\", \"content\": f\"hello world!\"}])\n",
     "r = openaimon.ChatCompletion.create(model=OPENAI_MODEL, messages=[{\"role\": \"user\", \"content\": f\"what is 2+2?\"}])"
```
Comment (staceysv) [inline examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb:305]: minor: should we add the standard text here about how the user can create a board/link them to something?

File: examples/prompts/trace_debugging/trace_quickstart_decorator.ipynb
Comment (staceysv) [inline examples/prompts/trace_debugging/trace_quickstart_decorator.ipynb:1]: nit: I get this install error, not sure why? is it importing weave and wandb in the same line perhaps?
everything runs fine.

 ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pandas-gbq 0.17.9 requires pyarrow<10.0dev,>=3.0.0, but you have pyarrow 13.0.0 which is incompatible.
tensorflow 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.8.0 which is incompatible.

File: examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb
Comment (staceysv) [inline examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb:1]: nit: install error here too, could it be something about my colab setup?
!pip install -qqq weave
!pip install -qqq weave
account_circle
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 26.6/26.6 MB 47.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 29.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 214.7/214.7 kB 9.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 24.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.0/40.0 MB 13.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.8/90.8 kB 9.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.1/65.1 kB 6.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 202.9/202.9 kB 20.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.2/65.2 kB 6.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.5/54.5 kB 5.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.1/143.1 kB 14.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 75.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 67.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 190.0/190.0 kB 17.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.7/62.7 kB 6.0 MB/s eta 0:00:00
  Building wheel for umap-learn (setup.py) ... done
  Building wheel for pynndescent (setup.py) ... done
  Building wheel for pathtools (setup.py) ... done
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pandas-gbq 0.17.9 requires pyarrow<10.0dev,>=3.0.0, but you have pyarrow 13.0.0 which is incompatible.
tensorflow 2.13.0 requires typing-extensions<4.6.0,>=3.6.6, but you have typing-extensions 4.8.0 which is incompatible.

**Review by staceysv — APPROVED** [review]
looks great, thanks so much Tim!

minor issues:
- installation of weave alone errors—prints out red errors/"restart runtime"—this looks bad but doesn't affect execution
- still an error in the openai_proxy_quickstart on the second CURL cell—made a note, not sure if expected.

File: examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb (hunk 3 above)
Comment (tssweeney) [reply examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb:285]: fixed in next commit

File: examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb (hunk 4 above)
Comment (tssweeney) [reply examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb:305]: this init monitor should dump out a link to the table. I will leave it to you to add more instructions if needed (:

File: examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb
Comment (tssweeney) [reply examples/prompts/trace_debugging/dev/synthetic_trace_data.ipynb:1]: this is scary looking, but not bad actually... we should fix this, but different PR.

File: examples/prompts/trace_debugging/trace_quickstart_decorator.ipynb
Comment (tssweeney) [reply examples/prompts/trace_debugging/trace_quickstart_decorator.ipynb:1]: same as below - fix in future pr

File: examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb
Comment (tssweeney) [reply examples/prompts/llm_monitoring/openai_proxy_quickstart.ipynb:1]: fixed in next commit

**Pull request merged**